creusot-contracts/src/lib.rs

```diff
@@ -162,6 +162,20 @@ pub mod logic {
             Ghost(std::marker::PhantomData)
         }
     }
+
+    pub struct Mapping<A: ?Sized, B: ?Sized>(std::marker::PhantomData<A>, std::marker::PhantomData<B>);
+
+    pub struct Seq<T: ?Sized>(std::marker::PhantomData<T>);
+
+    pub struct FSet<T: ?Sized>(std::marker::PhantomData<T>);
+
+    pub struct Set<T: ?Sized>(std::marker::PhantomData<T>);
+
+    pub struct Int;
```

Comment:
Can you instead remove the conditional flag on the relevant modules? They should compile just fine with `contracts` turned off (since `#[predicate]` and `#[logic]` will turn to no-ops.

Reply:
I tried doing that but it quite that easy, most types had `creusot::builtins = ...` attributes which needed to be disabled, the `EMPTY` constants also needed to be disabled since they had attributes but not `#[predicate]` or `#[logic]`, and implementations of standard library traits needed to be disabled since they did have `#[logic]` functions got removed leaving incomplete implementations.

Reply:
> I tried doing that but it quite that easy, most types had creusot::builtins = ... attributes which needed to be disabled, the EMPTY constants also needed to be disabled since they had attributes but not #[predicate] or #[logic], and implementations of standard library traits needed to be disabled since they did have #[logic] functions got removed leaving incomplete implementations.

Right, that's annoying.